src/main/kotlin/no/nav/klage/oppgave/api/KlagebehandlingFacade.kt

```diff
@@ -21,11 +30,40 @@ class KlagebehandlingFacade(
         private val securelogger = getSecureLogger()
     }
 
-    fun getKlagebehandling(oppgaveId: Long): KlagebehandlingView {
+    fun getKlagebehandling(klagebehandlingId: UUID): KlagebehandlingView {
         return klagebehandlingMapper.mapKlagebehandlingToKlagebehandlingView(
-            klagebehandlingService.getKlagebehandlingByOppgaveId(
-                oppgaveId
+            klagebehandlingService.getKlagebehandling(klagebehandlingId)
+        )
+    }
+
+    fun searchOppgaver(oppgaverSearchCriteria: OppgaverSearchCriteria): OppgaverRespons {
+        val esResponse = elasticsearchRepository.findByCriteria(oppgaverSearchCriteria)
+        return OppgaverRespons(
+            antallTreffTotalt = esResponse.totalHits.toInt(),
+            oppgaver = oppgaveMapper.mapEsKlagebehandlingerToView(
+                esResponse.searchHits.map { it.content },
+                oppgaverSearchCriteria.isProjectionUtvidet()
             )
         )
     }
+
+    fun assignOppgave(klagebehandlingId: UUID, saksbehandlerIdent: String?, klagebehandlingVersjon: Long) {
```

Comment:
Her gjør jeg assignment i vår egen database først (klagebehandling). Så prøver jeg å oppdatere Oppgave. Endringene i Oppgave vil så flyte tilbake til oss, men da stoppe i Mottak. Så en liten stund vil Klagebehandling og Mottak ikke være i sync, men så vil de bli det etterhvert.

Reply:
Kanskje et dumt spørsmål, men må vi oppdatere Oppgave her (som jeg nå tolker som 1. instans)? Det skader vel heller ikke så klart regner jeg med.

Reply:
Jeg tror det er bra nok. Jeg tror ikke vi skal overtenke problemet, selv om det kan oppstå. For oppgaven i Gosys skal på dette tidspunktet strengt tatt være assignet til vårt system, så jeg tror ikke dette skal gi problemer. Og dersom det kommer noen unntak der det skjer vil de være få og håndterbare.

Reply:
Det er ikke et dumt spørsmål Andreas, men jeg tror vi har litt ulike måter å tenke på. Jeg tenker det er greit at vi holder Oppgave oppdatert, i tilfelle noen er inne og ser i/jobber i Gosys. Spesielt på ting som assignment egentlig.